crates/resolver/src/system_conf/windows.rs

```diff
@@ -8,44 +8,64 @@
 //! System configuration loading for windows
 
 use std::net::{IpAddr, Ipv6Addr};
+use std::os::raw::c_ulong;
+use std::ptr;
 use std::str::FromStr;
 
-use ipconfig::computer::{get_domain, get_search_list};
-use ipconfig::get_adapters;
+use windows_registry::LOCAL_MACHINE;
+use windows_sys::Win32::Foundation::{
+    ERROR_BUFFER_OVERFLOW, ERROR_INVALID_PARAMETER, ERROR_NO_DATA, ERROR_NOT_ENOUGH_MEMORY,
+    ERROR_SUCCESS,
+};
+use windows_sys::Win32::NetworkManagement::IpHelper::{
+    GAA_FLAG_INCLUDE_GATEWAYS, GAA_FLAG_INCLUDE_PREFIX, GetAdaptersAddresses,
+    IP_ADAPTER_ADDRESSES_LH,
+};
+use windows_sys::Win32::Networking::WinSock::{
+    AF_INET, AF_INET6, AF_UNSPEC, SOCKADDR, SOCKADDR_IN, SOCKADDR_IN6, SOCKET_ADDRESS,
+};
 
 use crate::config::{NameServerConfig, ResolverConfig, ResolverOpts};
 use crate::proto::ProtoError;
 use crate::proto::rr::Name;
 
 pub fn read_system_conf() -> Result<(ResolverConfig, ResolverOpts), ProtoError> {
-    let adapters = get_adapters().map_err(|e| format!("ipconfig::get_adapters() failed: {e}"))?;
-
-    let servers = adapters
-        .iter()
-        // Only take interfaces whose OperStatus is Up
-        .filter(|adapter| adapter.oper_status() == ipconfig::OperStatus::IfOperStatusUp)
-        .flat_map(|adapter| adapter.dns_servers());
-
-    let mut name_servers = vec![];
-    for &ip in servers {
-        if let IpAddr::V6(ip) = ip {
-            if FORBIDDEN_ADDRS.contains(&ip) {
-                continue;
-            }
-        }
+    // Preallocate 16K per Microsoft recommendation, see Remarks section
+    // https://docs.microsoft.com/en-us/windows/desktop/api/iphlpapi/nf-iphlpapi-getadaptersaddresses
+    let mut buf_len = 16 * 1024u32;
 
-        name_servers.push(NameServerConfig::udp_and_tcp(ip));
-    }
+    let mut buffer = vec![0u8; buf_len as usize];
+    let result = unsafe {
+        GetAdaptersAddresses(
+            AF_UNSPEC as u32,
+            GAA_FLAG_INCLUDE_GATEWAYS | GAA_FLAG_INCLUDE_PREFIX,
+            ptr::null_mut(), // Reserved, must be null
+            buffer.as_mut_ptr() as *mut IP_ADAPTER_ADDRESSES_LH,
```

Comment:
I'm not exceptionally confident w/ `unsafe` but I _think_ `Vec<u8>` is 1-byte aligned and since `IP_ADAPTER_ADDRESSES_LH` is a `#[repr(C)]` struct (?) wouldn't we want to use something like `Layout::from_size_align(buf_len as usize, mem::align_of::<IP_ADAPTER_ADDRESSES_LH>())` to avoid UB?

The original `ipconfig` code doesn't do this, but I'm not sure I understand why not. Possible I'm wrong here?